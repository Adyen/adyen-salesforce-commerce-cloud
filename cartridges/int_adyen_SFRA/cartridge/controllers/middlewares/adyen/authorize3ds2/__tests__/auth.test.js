"use strict";

jest.mock('../payment');
jest.mock('../errorHandler', function () {
  return {
    toggle3DS2Error: jest.fn(),
    handlePaymentError: jest.fn()
  };
});

var createAuthorization = require("../auth");

var req;
var session;
var res;
var adyenCheckout;
var URLUtils;
var handlePaymentsCall;
var toggle3DS2Error;
beforeEach(function () {
  jest.clearAllMocks();
  req = {
    form: {
      resultCode: 'Authorized',
      stateData: '{ "details":"mocked details"}'
    },
    locale: {
      id: 'mocked_locale'
    }
  };
  res = {
    redirect: jest.fn()
  };
  handlePaymentsCall = require('../payment');
  toggle3DS2Error = require('../errorHandler').toggle3DS2Error;
});
describe('Auth', function () {
  it('should handle 3ds2 auth when challenge shopper', function () {
    req.form.resultCode = 'ChallengeShopper';
    req.form.challengeResult = true;
    createAuthorization({
      req: req,
      res: res,
      next: jest.fn()
    });
    expect(handlePaymentsCall.mock.calls).toMatchSnapshot();
  });
  it('should handle 3ds2 auth when has fingerprint', function () {
    req.form.resultCode = 'IdentifyShopper';
    req.form.fingerprintResult = true;
    createAuthorization({
      req: req,
      res: res,
      next: jest.fn()
    });
    expect(handlePaymentsCall.mock.calls).toMatchSnapshot();
  });
  it('should handle 3ds2 error', function () {
    req.form.resultCode = 'NOT_AUTHORIZED';
    createAuthorization({
      req: req,
      res: res,
      next: jest.fn()
    });
    expect(toggle3DS2Error.mock.calls).toMatchSnapshot();
  });
});