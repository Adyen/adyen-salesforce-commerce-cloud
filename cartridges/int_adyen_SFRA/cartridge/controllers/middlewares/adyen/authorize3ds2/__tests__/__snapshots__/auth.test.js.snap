// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should handle 3ds2 auth when challenge shopper 1`] = `
Array [
  Array [
    Object {
      "details": "mocked details",
      "paymentData": "mocked_adyen_payment_data",
    },
    Object {
      "custom": Object {
        "Adyen_pspReference": "mocked_pspRef",
      },
      "getPaymentInstruments": [MockFunction] {
        "calls": Array [
          Array [
            "mocked_method",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Object {
              "0": Object {
                "custom": Object {
                  "adyenAction": "mocked_adyen_action",
                  "adyenMD": "mocked_adyen_MD",
                  "adyenPaymentData": "mocked_adyen_payment_data",
                  "adyenRedirectURL": "https://some_mocked_url/signature",
                },
              },
              "iterator": [Function],
              "toArray": [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": Array [
                      Object {
                        "custom": Object {
                          "adyenAction": "mocked_adyen_action",
                          "adyenMD": "mocked_adyen_MD",
                          "adyenPaymentData": "mocked_adyen_payment_data",
                          "adyenRedirectURL": "https://some_mocked_url/signature",
                        },
                      },
                    ],
                  },
                ],
              },
            },
          },
        ],
      },
      "orderNo": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
      "setExportStatus": [MockFunction],
      "setPaymentStatus": [MockFunction],
    },
    Object {
      "custom": Object {
        "adyenAction": "mocked_adyen_action",
        "adyenMD": "mocked_adyen_MD",
        "adyenPaymentData": "mocked_adyen_payment_data",
        "adyenRedirectURL": "https://some_mocked_url/signature",
      },
    },
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "challengeResult": true,
          "resultCode": "ChallengeShopper",
          "stateData": "{ \\"details\\":\\"mocked details\\"}",
        },
        "locale": Object {
          "id": "mocked_locale",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;

exports[`Auth should handle 3ds2 auth when has fingerprint 1`] = `
Array [
  Array [
    Object {
      "details": "mocked details",
      "paymentData": "mocked_adyen_payment_data",
    },
    Object {
      "custom": Object {
        "Adyen_pspReference": "mocked_pspRef",
      },
      "getPaymentInstruments": [MockFunction] {
        "calls": Array [
          Array [
            "mocked_method",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Object {
              "0": Object {
                "custom": Object {
                  "adyenAction": "mocked_adyen_action",
                  "adyenMD": "mocked_adyen_MD",
                  "adyenPaymentData": "mocked_adyen_payment_data",
                  "adyenRedirectURL": "https://some_mocked_url/signature",
                },
              },
              "iterator": [Function],
              "toArray": [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": Array [
                      Object {
                        "custom": Object {
                          "adyenAction": "mocked_adyen_action",
                          "adyenMD": "mocked_adyen_MD",
                          "adyenPaymentData": "mocked_adyen_payment_data",
                          "adyenRedirectURL": "https://some_mocked_url/signature",
                        },
                      },
                    ],
                  },
                ],
              },
            },
          },
        ],
      },
      "orderNo": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
      "setExportStatus": [MockFunction],
      "setPaymentStatus": [MockFunction],
    },
    Object {
      "custom": Object {
        "adyenAction": "mocked_adyen_action",
        "adyenMD": "mocked_adyen_MD",
        "adyenPaymentData": "mocked_adyen_payment_data",
        "adyenRedirectURL": "https://some_mocked_url/signature",
      },
    },
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "fingerprintResult": true,
          "resultCode": "IdentifyShopper",
          "stateData": "{ \\"details\\":\\"mocked details\\"}",
        },
        "locale": Object {
          "id": "mocked_locale",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;

exports[`Auth should handle 3ds2 error 1`] = `
Array [
  Array [
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "resultCode": "NOT_AUTHORIZED",
          "stateData": "{ \\"details\\":\\"mocked details\\"}",
        },
        "locale": Object {
          "id": "mocked_locale",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;
